src/plugins/legacy/polygonRoi/polygonRoiToolBox.cpp

```diff
@@ -163,122 +145,216 @@ medAbstractData *polygonRoiToolBox::processOutput()
 
         if (!viewEventFilter)
         {
             viewEventFilter = new polygonEventFilter(currentView);
-            connect(viewEventFilter, SIGNAL(enableRepulsor(bool)), this, SLOT(enableRepulsorButton(bool)), Qt::UniqueConnection);
-            connect(viewEventFilter, SIGNAL(enableGenerateMask(bool)), this, SLOT(enableGenerateMaskButton(bool)), Qt::UniqueConnection);
+            connect(viewEventFilter, SIGNAL(enableRepulsor(bool)), repulsorTool, SLOT(setEnabled(bool)), Qt::UniqueConnection);
+            connect(viewEventFilter, SIGNAL(enableGenerateMask(bool)), saveBinaryMaskButton, SLOT(setEnabled(bool)), Qt::UniqueConnection);
             connect(viewEventFilter, SIGNAL(enableViewChooser(bool)), this, SLOT(enableTableViewChooser(bool)), Qt::UniqueConnection);
+            connect(viewEventFilter, SIGNAL(toggleRepulsorButton(bool)), this, SLOT(activateRepulsor(bool)), Qt::UniqueConnection);
         }
+
+        viewEventFilter->updateView(currentView);
         viewEventFilter->On();
-        medViewContainer* container = nullptr;
-        foreach (QUuid uuid, this->getWorkspace()->tabbedViewContainers()->containersSelected())
+        viewEventFilter->installOnView(currentView);
+
+        if ( viewEventFilter->isContourInSlice() )
         {
-            container = medViewContainerManager::instance()->container(uuid);
-            if(container)
+            repulsorTool->setEnabled(state);
+            if (repulsorTool->isChecked())
             {
-                viewEventFilter->installOnView(container->view());
+                viewEventFilter->activateRepulsor(state);
             }
         }
+        connect(viewEventFilter, SIGNAL(clearLastAlternativeView()), this, SLOT(resetToolboxBehaviour()));
     }
     else
     {
+        repulsorTool->setEnabled(state);
         viewEventFilter->Off();
     }
 }
 
 void polygonRoiToolBox::activateRepulsor(bool state)
 {
     if (currentView && viewEventFilter)
-        viewEventFilter->activateRepulsor(state);
+    {
+        if (state)
+        {
+            if ( viewEventFilter->isContourInSlice())
+            {
+                viewEventFilter->activateRepulsor(state);
+            }
+            else
+            {
+                repulsorTool->setChecked(false);
+            }
+        }
+        else
+        {
+            repulsorTool->setChecked(false);
+            viewEventFilter->activateRepulsor(state);
+        }
+    }
+}
+
+void polygonRoiToolBox::copyContours()
+{
+//    if (viewEventFilter)
+//        viewEventFilter->copyContours();
+}
+
+void polygonRoiToolBox::pasteContours()
+{
+//    if (viewEventFilter)
+//        viewEventFilter->pasteContours();
+}
+
+void polygonRoiToolBox::resetToolboxBehaviour()
+{
+    medTabbedViewContainers *containers = this->getWorkspace()->tabbedViewContainers();
+    QList<medViewContainer*> containersInTabSelected = containers->containersInTab(containers->currentIndex());
+    if (containersInTabSelected.size() != 1)
```

Comment:
there are some {} that you can add in the changes in this file

Reply:
done